pkg/virt-operator/resource/generate/components/secrets.go

```diff
@@ -273,7 +272,21 @@ func NextRotationDeadline(cert *tls.Certificate, ca *tls.Certificate, duration *
 	}
 
 	notAfter := cert.Leaf.NotAfter
-	deadline := notAfter.Add(-time.Duration(float64(duration.Duration) * 0.2))
+	caDeadline := notAfter
+	deadline := cert.Leaf.NotBefore.Add(renewBefore.Duration)
```

Comment:
@stu-gott I have 2 questions about this calulation : 1. Is it possible that `renewBefore` can be smaller then `cert.Leaf.NotBefore`? 2. In case the `renewBefore` is bigger then the `NotBefore`, shouldn't the deadline be `cert.Leaf.NotBefore.Add((renewBefore - cert.Leaf.NotBefore).Duration)`?

Reply:
OK. I had to dig to prove to myself what would happen if the renewBefore was nonsense. it turns out this is a valid concern. the "duration" plays into the "NotAfter" field, but "renewBefore" isn't even a concept in x509--that's entirely for our accounting/scheduling that this value exists.

Thus it makes sense to account for a renewBefore that's just too late. Sadly the most recent changeset removed the duration field. :D I propose this: I'll back that function out so that we have the duration available here. If "renewBefore" is longer than the certificate duration, I think the only sane response is to fall back to the previous behavior where the "renewBefore" value didn't exist and we just calculated "80%" of the duration. What do you think?

Reply:
@stu-gott Sounds good to me. Another option which may be more strict is to validate these values in the validating webhook and not allow a `duration < renewBefore` situation. And maybe even set a minimum interval between these two

Reply:
@stu-gott Another nonsense scenario can be when `cert.renewBefore > ca.RenewBefore`. That actually means that server cert renewal will occur because of the CA renewal, so `cert.renewBefore` will turn meaningless.

Reply:
Actually, this isn't a concern.

This same code path is used for both CA's and Certs--so we know the CA's renewBefore occurs before it's duration. This code already checks that the Cert must be refreshed before the CA becomes invalid. Remember, there's an overlap period where both the outgoing and the incoming CA's are honored. As long as our cert is rotated before *that* cutoff, things remain sane.

Reply:
As far as the validating webhook-there isn't one. KubeVirt resources are created by operators other than KubeVirt itself.

Reply:
@stu-gott I see that updates can be validated after [4599](https://github.com/kubevirt/kubevirt/pull/4599) got merged.

Reply:
:facepalm: I didn't know that was there. testing a patch for this now.

Reply:
fixed!